workflows/checks/check-md5.nf

```diff
@@ -13,6 +13,11 @@ params.run_ids = "SCPCR000001,SCPCR000002"
 process check_md5{
   container 'ubuntu:20.04'
   label 'bigdisk'
+  // tasks sometimes fail due to disk space but we can't specify that directly
+  errorStrategy 'retry' 
+  maxRetries = 1
+  cpus { 1 + 3 * (task.attempt - 1)  } //use more CPUs to take over a machine if failure
```

Comment:
Is `task.attempt` initialized elsewhere? If so, where? If not, can you explain why it works here without being defined or initialized elsewhere?

Also, if I understand the above equation correctly, does this mean that the process will only be retried once if needed, and on that _second_ try, we will use (1 + 3) times more CPUs than (task.attempt - 1) which would be (2 -1). Is that correct?

Reply:
`task.attempt` is provided by Nextflow, and reflects task attempt count, so it is just 1, 2, 3, etc. (though since we only retry once, we can't get `task.attempt` > 2). The idea is that the first time it will use 1 cpu, the second time it will use 4, which is usually a full AWS machine, but we don't directly control that, so it may not always work 🤷🏼

Reply:
Ahh gotcha! 👍